studiopress/atomic-blocks
# Add/auto installation of genesis blocks

This PR builds on the Migrate admin page in #405, and hooks up the "Install Genesis Blocks" button to install Genesis Blocks using ajax and the REST API. 

If it fails, it shows an error message and the option to retry. 

If it successfully installs, it redirects to the Genesis Blocks Migration page being built in https://github.com/studiopress/genesis-blocks/pull/112

### How to test
0. Make sure you do not have Genesis Blocks installed.
1. In wp-admin, under "Atomic Blocks" click "Migrate".
2. Once the page loads, disconnect from the internet.
3. Click the "Install Genesis Blocks" button. 
4. Make sure you get a failure notice that "Something went wrong" and the option to retry installing. 
5. Reconnect to the internet. 
6. Comment out the nonce in ab-migrate.js on line 10 to simulate a timed-out nonce.
7. Click the "Install Genesis Blocks" button. 
8. Make sure you get a failure notice that "You are not allowed to do that".
9. Undo what you did in step 6. 
10. Click the "Install Genesis Blocks" button. 
11. A spinner should show while you wait.
12. Upon successful installation, the spinner should hide, and you should see `Successfully installed and activated Genesis Blocks!`.
13. After 1 second, you should be automatically redirected to the [migrate page in Genesis Blocks](https://github.com/studiopress/genesis-blocks/pull/112). Note that at the moment this will show `Sorry, you are not allowed to access this page.` because that page does not yet exist in the version of Genesis Blocks that is pulled from the wp.org server.
